sympy/physics/mechanics/tests/test_joint.py

```diff
@@ -0,0 +1,301 @@
+    assert child_joint_point.vel(child.frame) == Vector(0)
+    assert child_joint_point.vel(parent.frame) == vel * parent.frame.x
+    assert child.masscenter.vel(parent.frame) == vel * parent.frame.x
+
+
+def test_sliding_joint_arguments():
+    parent = Body('parent')
+    child = Body('child')
+    l1 = Symbol('l1')
+    l2 = Symbol('l2')
+    sliding_joint = SlidingJoint('sliding_joint', parent, child,
+                                 parent_point_pos=(0, l1, 0),
+                                 child_point_pos=(0, l2, 0),
+                                 parent_axis=parent.frame.z,
+                                 child_axis=child.frame.y)
+    assert sliding_joint.name == 'sliding_joint'
+    assert sliding_joint.parent == parent
+    assert sliding_joint.child == child
+    assert sliding_joint._parent_joint_location == l1 * parent.frame.y
+    assert sliding_joint._child_joint_location == l2 * child.frame.y
+    assert sliding_joint.parent_axis == parent.frame.z
+    assert sliding_joint.child_axis == child.frame.y
+
+
+def test_cylindrical_joint():
+    parent = Body('parent')
+    child = Body('child')
+    l = Symbol('l')
+    cylindrical_joint = CylindricalJoint('cylindrical_joint', parent, child,
+                                         child_point_pos=(l, l, 0))
+
+    assert cylindrical_joint.name == 'cylindrical_joint'
+    assert cylindrical_joint.parent == parent
+    assert cylindrical_joint.child == child
+    assert cylindrical_joint._parent_joint_location == Vector(0)
+    assert cylindrical_joint._child_joint_location == l * (child.frame.x +
+                                                        child.frame.y)
+    assert cylindrical_joint.parent_axis == parent.frame.x
+    assert cylindrical_joint.child_axis == child.frame.x
+
+    dis = dynamicsymbols('cylindrical_joint_dis')
+    disd = dynamicsymbols('cylindrical_joint_dis', 1)
+    vel = dynamicsymbols('cylindrical_joint_vel')
+
+    theta = dynamicsymbols('cylindrical_joint_theta')
+    thetad = dynamicsymbols('cylindrical_joint_theta', 1)
+    omega = dynamicsymbols('cylindrical_joint_omega')
+
+    assert cylindrical_joint.coordinates == [dis, theta]
+    assert cylindrical_joint.speeds == [vel, omega]
+    assert cylindrical_joint.kds == [disd - vel, thetad - omega]
+
+    child_joint_point = cylindrical_joint.child_joint_point
+    parent_joint_point = cylindrical_joint.parent_joint_point
+    assert parent_joint_point.name == 'cylindrical_joint_parent_joint'
+    assert child_joint_point.name == 'cylindrical_joint_child_joint'
+    # assert child_joint_point.pos_from(parent_joint_point) == dis * parent.frame.x
+    child_joint_masscenter = l * (child.frame.x + child.frame.y)
+    assert child_joint_point.pos_from(child.masscenter) == child_joint_masscenter
+    assert parent_joint_point.pos_from(parent.masscenter) == Vector(0)
+    assert child.frame.ang_vel_in(parent.frame) == omega * parent.frame.x
+
```

Comment:
This needs to have checks for orientations (director cosine matrices) and linear velocities of points.